lib/ddtrace/sampling/rule_sampler.rb

```diff
@@ -0,0 +1,56 @@
+require 'ddtrace/ext/priority'
+
+require 'ddtrace/ext/sampling'
+require 'ddtrace/sampler'
+
+module Datadog
+  module Sampling
+    # TODO:
+    class RuleSampler
+      extend Forwardable
+
+      attr_reader :rules, :rate_limiter, :priority_sampler
+
+      def initialize(rules, rate_limiter, priority_sampler = Datadog::RateByServiceSampler.new)
```

Comment:
we would need to make sure this priority sampler gets it's rates updated from responses from the agent.

Reply:
That's taken take by delegating `:update` to that sampler, like so further down this class:
```ruby
def_delegators :@priority_sampler, :update
```